docs/markdown/guides/mermaid/index.ts

```diff
@@ -0,0 +1,62 @@
+import {DocsCodeToken} from '../extensions/docs-code/docs-code';
+import {chromium} from 'playwright-core';
+import {createRequire} from 'node:module';
+import {Mermaid, MermaidConfig} from 'mermaid';
+import {runfiles} from '@bazel/runfiles';
+import {join} from 'node:path';
+import {mermaidStyling} from './styling';
+
+// Declare mermarid in the context of this file so that typescript doesn't get upset when we
+// access it within the `page.evaluate` function. At runtime the context in with the method
+// is run difference than this file, but this makes typescript happy.
+declare const mermaid: Mermaid;
+
+/** Mermaid configuration to use when creating mermaid svgs. */
+const mermaidConfig: MermaidConfig = {
+  ...mermaidStyling,
+};
+
+/** The full path to the mermaid script. */
+let mermaidScriptTagData: {path: string} | undefined;
+
+/** Get the mermaid script file path, resolving it if necessary first. */
+function getMermaidScriptTagData() {
+  if (mermaidScriptTagData) {
+    return mermaidScriptTagData;
+  }
+  const require = createRequire(import.meta.url);
+  return (mermaidScriptTagData = {path: require.resolve('mermaid/dist/mermaid.js')});
+}
+
+/** Replace the code block content with the mermaid generated SVG element string in place. */
+export async function processMermaidCodeBlock(token: DocsCodeToken) {
+  /** Browser instance to run mermaid within. */
+  const browser = await chromium.launch({
+    headless: true,
+    executablePath: join(runfiles.resolveWorkspaceRelative('.'), process.env['CHROME_BIN']!),
+    args: ['--no-sandbox'],
+  });
+  /** Page to run mermaid in. */
+  const page = await browser.newPage({bypassCSP: true});
```

Comment:
Nit: Add a comment on why `bypassCSP`

Reply:
I removed it, I was running into some issues at one point I think because of injecting the script tag after initial load, but it seems to work now.